src/hotspot/share/runtime/interfaceSupport.inline.hpp

```diff
@@ -229,14 +229,12 @@ class ThreadBlockInVMPreprocess : public ThreadStateTransition {
   ~ThreadBlockInVMPreprocess() {
     assert(_thread->thread_state() == _thread_blocked, "coming from wrong thread state");
     // Change to transition state and ensure it is seen by the VM thread.
-    _thread->set_thread_state_fence(_thread_blocked_trans);
+    _thread->set_thread_state_fence(_thread_in_vm);
```

Comment:
The comment on L231 is now wrong since you're no longer using
a "transition" state. Perhaps, "Change to an unsafe state..."

Reply:
Fixed.